### PR TITLE
Shared: Update node settings

### PR DESCRIPTION
**Pull request by cvarley100**
## Description

- Fixes various issues with setting a specific node for remote PoW
- Removes PoW-only settings page
- Updates node settings UI and UX
- Adds specific PoW node auto-selection setting

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

Tested on iOS and Mac

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [ ] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [ ] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: src/desktop/src/ui/views/settings/Node.js [hunk 1]
```diff
@@ -118,12 +118,15 @@ const NodeSettings = ({ customNodes, generateAlert, loading, nodes, settings, ac
         if (quorumEnabled !== settings.quorumEnabled || quorumSize !== settings.quorumSize) {
             actions.updateQuorumConfig({ enabled: quorumEnabled, size: quorumSize });
         }
+        if (powNode !== settings.powNode) {
+            actions.setPowNode(powNode);
+        }
+        if (powNode.length ^ this.props.powNode.length) {
```
Comment (rajivshah3) [inline src/desktop/src/ui/views/settings/Node.js]:
```suggestion
        if (powNode.length ^ settings.powNode.length) {
```

>124:30  error  Stateless functional components should not use this  react/no-this-in-sfc

https://buildkite.com/iota-foundation/trinity-desktop-prs/builds/4935#8a8cb4cb-3e4f-495a-93f3-f5a1f8ff734f/54-259

File: src/desktop/src/ui/views/settings/Node.js [hunk 2]
```diff
@@ -118,12 +118,15 @@ const NodeSettings = ({ customNodes, generateAlert, loading, nodes, settings, ac
...
+            this.props.changePowSettings();
```
Comment (rajivshah3) [inline src/desktop/src/ui/views/settings/Node.js]:
```suggestion
            actions.changePowSettings();
```

>125:13  error  Stateless functional components should not use this  react/no-this-in-sfc

https://buildkite.com/iota-foundation/trinity-desktop-prs/builds/4935#8a8cb4cb-3e4f-495a-93f3-f5a1f8ff734f/54-260

File: src/mobile/src/ui/views/wallet/NodeSettings.js [hunk 3]
```diff
@@ -245,6 +266,17 @@ export class NodeSettings extends PureComponent {
...
     }
 
+    /**
+     * Toggles proof of work between local and remote and sets node to outsource to
+     *
+     * @method togglePow
+     */
+     togglePowNode(){
+         const availablePowNodes = this.getAvailableNodes(true);
+         const powNode = this.state.powNode.length === 0 && availablePowNodes.length ? availablePowNodes[0].url : '';
```
Comment (rajivshah3) [inline src/mobile/src/ui/views/wallet/NodeSettings.js]: >276:26  error  Use callback in setState when referencing the previous state  react/no-access-state-in-setstate

https://buildkite.com/iota-foundation/trinity-mobile-prs/builds/4986#647b529d-0083-4b4e-9dfa-f981c8d7f627/54-297

**Review by rajivshah3 — CHANGES_REQUESTED** [review]

File: src/shared/libs/iota/NodesManager.js [hunk 4]
```diff
@@ -84,9 +92,17 @@ export default class NodesManager {
...
-            if (!nodeAutoSwitch) {
-                return (...args) => promiseFunc(primaryNode, quorum.enabled)(...args);
+            // If powNode is defined, then this is an attachToTangle request
+            if (powNode) {
+                // If auto pow node switching is disabled, just connect to the selected (pow) node
+                if (!powNodeAutoSwitch) {
+                    return (...args) => promiseFunc(powNode, quorum.enabled)(...args);
+                }
+            } else {
+                // If auto primaru node switching is disabled, just connect to the selected (primary) node
```
Comment (rajivshah3) [inline src/shared/libs/iota/NodesManager.js]:
```suggestion
                // If auto primary node switching is disabled, just connect to the selected (primary) node
```

File: src/shared/locales/en/translation.json [hunk 5]
```diff
@@ -501,9 +501,9 @@
     },
     "nodeSettings": {
         "automaticNodeManagement": "Automatic node management",
-        "addCustomNodes": "Add custom nodes",
-        "autoNodeList": "Use remote node list",
-        "nodeAutoswitching": "Primary node autoswitching",
+        "addCustomNodes": "Add nodes",
+        "autoNodeList": "Use official node list",
```
Comment (rajivshah3) [inline src/shared/locales/en/translation.json:505]: Would `default` sound better here?

**Review by rihardsgravis — APPROVED** [review]

**Pull request merged**